v2/test/pre-release/pre_release_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package multitenant_test
+
+import (
+	"fmt"
+	"testing"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1beta20201101"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	resourceGroupName     = "upgrade-test-rg"
+	preReleaseNamespace   = "pre-release"
+	vnetBeforeUpgradeName = "vnet-before-upgrade"
+)
+
+func Test_Pre_Release_ResourceCanBeCreated_BeforeUpgrade(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+	tc.Namespace = preReleaseNamespace
+
+	rg := tc.NewTestResourceGroup()
+	rg.Name = getLiveTestResourceName(resourceGroupName)
+
+	vnet := newVnet(tc, getLiveTestResourceName(vnetBeforeUpgradeName))
+	// We want to use 'CreateResourceAndWaitWithoutCleanup' here as we don't want the resource cleanup for this test. We will be checking the
+	// backward compatibility for these resources after we upgrade the controller.
+	tc.CreateResourceAndWaitWithoutCleanup(rg)
+	tc.CreateResourceAndWaitWithoutCleanup(vnet)
+}
+
+func newVnet(tc *testcommon.KubePerTestContext, name string) *network.VirtualNetwork {
+	vnet := &network.VirtualNetwork{
+		ObjectMeta: tc.MakeObjectMetaWithName(name),
+		Spec: network.VirtualNetworks_Spec{
+			Owner: &genruntime.KnownResourceReference{
+				Name: getLiveTestResourceName(resourceGroupName),
+			},
+			Location: tc.AzureRegion,
+			AddressSpace: &network.AddressSpace{
+				AddressPrefixes: []string{"10.0.0.0/8"},
+			},
+		},
+	}
+	return vnet
+}
+
+func Test_Pre_Release_ResourceCanBeCreated_AfterUpgrade(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	tc.Namespace = preReleaseNamespace
+
+	// This resource already will exist in kind as will be created without cleanup in test 'Test_Pre_Release_ResourceCanBeCreated_BeforeUpgrade'.
+	vnetBeforeUpgrade := newVnet(tc, getLiveTestResourceName(vnetBeforeUpgradeName))
+	defer tc.DeleteResourceAndWait(vnetBeforeUpgrade)
+
+	// TODO: Will have to change the version here when we go from beta to stable
+	vnetAfterUpgrade := newVnet(tc, tc.Namer.GenerateName("vn"))
+	tc.CreateResourceAndWait(vnetAfterUpgrade)
+	tc.DeleteResourcesAndWait(vnetAfterUpgrade)
+
+	// Patch for existing resource.
+	old := vnetBeforeUpgrade.DeepCopy()
+	vnetBeforeUpgrade.Spec.Tags = map[string]string{"newTag": "afterUpgradeTag"}
+	tc.Patch(old, vnetBeforeUpgrade)
+	tc.Eventually(vnetBeforeUpgrade).Should(tc.Match.BeProvisioned(1))
+	tc.GetResource(types.NamespacedName{Name: vnetBeforeUpgrade.Name, Namespace: vnetBeforeUpgrade.Namespace}, vnetBeforeUpgrade)
+	tc.Expect(vnetBeforeUpgrade.Status.Tags).To(gomega.HaveKey("newTag"))
+}
+
+func getLiveTestResourceName(name string) string {
+	return fmt.Sprintf("%s-%s", testcommon.LiveResourcePrefix, name)
```

Comment:
So technically, you could do this by passing `testcommon.LiveResourcePrefix` to `nameConfig` on line 39 of the `suite_test.go`, and then creating a custom namer in the test via:
`newNamer := tc.Namer.WithNumRandomChars(0)`.

Then you can do `newNamer.GenerateName(resourceGroupName)`. to generate a name that is the `resourceGroupName` + just the prefix.

The only advantage I see to doing it that way is that you don't need this local function to do that and it mirrors the way the live tests do it so that if there's any change in theory this place would also get it.

Pretty minor though.